cachito/workers/tasks.py

```diff
@@ -90,3 +88,39 @@ def assemble_source_code_archive(app_archive_path, deps_path, bundle_archive_pat
                 bundle_archive.addfile(member, app_archive.extractfile(member.name))
 
         bundle_archive.add(absolute_deps_path, 'deps')
+
+
+@app.task
+def set_request_state(request_id, state, state_reason=None):
+    """
+    Set the state of the request using the Cachito API.
+
+    :param int request_id: the ID of the request
+    :param str state: the state to set the request to
+    :param str state_reason: the optional state reason to set the request to
+    :raise CachitoError: if the request to the Cachito API fails
+    """
+    config = get_worker_config()
+    request_url = f'{config.cachito_api_url.rstrip("/")}/requests/{request_id}'
+
+    log.info('Setting the status of request %d to "%s"', request_id, state)
+    payload = {'state': state}
+    if state_reason:
+        payload['state_reason'] = state_reason
+    try:
+        rv = requests.patch(request_url, json=payload, timeout=30)
```

Comment:
Let's add a config for timeout here.

Reply:
The request for downloading the source archive is set to 120. That seems reasonable, but this shouldn't be that high. So that means we'd have to have separate config options for both.

I like the idea of making this configurable, but let's make it a separate PR.